mypy/semanal.py

```diff
@@ -3083,6 +3083,18 @@ def lookup_qualified(self, name: str, ctx: Context,
                         n = names.get(parts[i], None)
                         if n and isinstance(n.node, ImportedName):
                             n = self.dereference_module_cross_ref(n)
+                        elif '__getattr__' in names:
+                            getattr_defn = names['__getattr__']
+                            if isinstance(getattr_defn.node, (FuncDef, Var)):
+                                if isinstance(getattr_defn.node.type, CallableType):
+                                    typ = getattr_defn.node.type.ret_type
+                                else:
+                                    typ = AnyType(TypeOfAny.from_error)
+                                name = parts[i]
+                                v = Var(name, type=typ)
+                                v._fullname = name
+                                n = SymbolTableNode(GDEF, v)
+                                names[name] = n
```

Comment:
It might be worth pulling this out into another method, since it is a big chunk of code for an obscure case.

Reply:
Makes sense. I there is also similar code above in `visit_import_from`, it would make sense to combine the two into a helper method. I will do this tomorrow.